x/staking/src/keeper/validator.rs

```diff
@@ -103,22 +265,22 @@ impl<
         if new_addrs.is_empty() {
             self.delete_validator_queue_time_slice(
                 ctx,
-                validator.unbonding_time,
+                validator.unbonding_time.clone(),
                 validator.unbonding_height,
             );
         } else {
             self.set_unbonding_validators_queue(
                 ctx,
-                validator.unbonding_time,
+                validator.unbonding_time.clone(),
                 validator.unbonding_height,
                 new_addrs,
-            )?;
+            );
         }
         Ok(())
     }
 
     /// get the last validator set
-    pub fn get_last_validators_by_addr<DB: Database, CTX: QueryableContext<DB, SK>>(
+    pub fn last_validators_by_addr<DB: Database, CTX: QueryableContext<DB, SK>>(
```

Comment:
Can this fail?

Reply:
We perform a conversion into `ValAddress`. I think we can use `unwrap_or_corrupt` because invalid conversion says that the database is broken. 
@joneskm Are you agree with that?

Reply:
Yes, should be unwrap_or_corrupt. Why do the validator addresses get converted to strings - we've lost type information by doing this?

Reply:
fixed